packages/typespec-test/test/openai_modular/generated/typespec-ts/review/openai_modular.api.md

```diff
@@ -115,12 +115,12 @@ export interface AzureChatExtensionConfiguration {
     type: AzureChatExtensionType;
 }
 
-// @public
+// @public (undocumented)
 export type AzureChatExtensionConfigurationUnion = AzureSearchChatExtensionConfiguration | AzureMachineLearningIndexChatExtensionConfiguration | AzureCosmosDBChatExtensionConfiguration | ElasticsearchChatExtensionConfiguration | PineconeChatExtensionConfiguration | AzureChatExtensionConfiguration;
 
 // @public
 export interface AzureChatExtensionDataSourceResponseCitation {
-    chunkId?: string;
+    chunk_id?: string;
```

Comment:
need to check which one is correct.

Reply:
it's align with the typespec definition.

Reply:
Do we have the ability to raise this as a warning? I like the idea of doing what is asked of us without trying to be smart, but I also like making sure the author is conscious of this deviation. 

/cc @lmazuel @bterlson

Reply:
With the exception of RLC, we have been trying to generate property names that align with common JS conventions. In this case I would expect we generate camelCase and transform back to the original in serialization.

My understanding of clientName is that each language would format it accordingly to their best practices and conventions, which is why I would also expect us to camelCase a clientName cliet_id as clientId.

the situations where Id worry about the emitter making assumptions is if we were changing semantics or naming anonymous things, and exposing to the public api.

But I think changing the casing to align with common JS conventions should be safe and desired/expected of the emitter.

Reply:
fixed in https://github.com/Azure/autorest.typescript/pull/2759/commits/c79d42e6e75af913d10c921fad08c07e622b01b8